io/net/virt-net/network_virtualization.py

```diff
@@ -420,6 +423,44 @@ def test_vnic_auto_failover(self):
         else:
             self.cancel("Provide more backing device, only 1 given")
 
+    def test_rmdev_viosfailover(self):
+        '''
+        using mrdev and mkdev command to check vios failover
+        '''
+        self.login(self.vios_ip, self.vios_user, self.vios_pwd)
+        cmd = "lsmap -all -vnic -cpid %s" % self.lpar_id
+        vnc_server = self.run_command(cmd)[2].split()[0]
+        cmd = "lsmap -vnic -vadapter %s" % vnc_server
+        output = self.run_command(cmd)
+        for line in output:
+            if 'Backing device' in line:
+                vnic_backing_device = line.split(':')[-1]
+        tmp = 'rmdev -l %s' % vnc_server
+        cmd = "echo \"%s\" | oem_setup_env" % tmp
+        output = self.run_command(cmd)
+        if "Defined" not in str(output):
+            self.fail("command fail in vios")
+        tmp = 'rmdev -l %s' % vnic_backing_device
+        cmd = "echo \"%s\" | oem_setup_env" % tmp
+        output = self.run_command(cmd)
+        if "Defined" not in str(output):
+            self.fail("command fail in vios")
+        time.sleep(20)
+        tmp = 'mkdev -l %s' % vnic_backing_device
+        cmd = "echo \"%s\"| oem_setup_env" % tmp
+        output = self.run_command(cmd)
+        if "Available" not in str(output):
+            self.fail("command fail in vios")
+        tmp = 'mkdev -l %s' % vnc_server
+        cmd = "echo \"%s\" | oem_setup_env" % tmp
+        output = self.run_command(cmd)
+        if "Available" not in str(output):
+            self.fail("command fail in vios")
```

Comment:
can be done better with helper functions

Reply:
yeah move this check to a function

Reply:
for each command..?

Reply:
@smruti77 There can be one function to `add/remove`    `vnic server/backing device`. See if you can achieve that.